app/src/main/java/com/edulectronics/tinycircuit/Models/Graph.java

```diff
@@ -0,0 +1,94 @@
+package com.edulectronics.tinycircuit.Models;
+
+import com.edulectronics.tinycircuit.Models.Components.Connectors.Connection;
+import com.edulectronics.tinycircuit.Models.Components.Powersource;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Stack;
+
+/**
+ * Created by Maaike on 2-1-2017.
+ */
+
+public class Graph {
+
+    private final Powersource source;
+    private final Object base;
+
+    public Set<Object> nodes = new HashSet<>();
+    public List<Edge> edges = new ArrayList<Edge>();
+
+    Stack connectionPath;
+    List<Stack> connectionPaths;
+
+    public Graph(Powersource source, ArrayList<Connection> connections) {
+        this.source = source;
+        this.base = new Object();
+
+        nodes.add(source);
+        nodes.add(base);
+
+        for (Connection connection: connections) {
+
+            Edge edge = new Edge(connection.pointA.getParentComponent(), connection.pointB.getParentComponent());
+
+            // Relay all source inputs to an object named 'base'. This way we can check
+            // all paths between source and base, even though they really go from source to source.
+            if(edge.a == source || edge.b == source) {
+                if(source.getInput() == connection.pointA) {
+                    edge.a = base;
+                }
+                if( source.getInput() == connection.pointB) {
```

Comment:
Spatie, misschien is het handig om af en toe Ctrl+Alt+L in te drukken, want dat lost dit soort esthetisch onverantwoordelijke zaken op